prov/sm2/src/sm2_progress.c

```diff
@@ -97,6 +97,7 @@ static int sm2_start_common(struct sm2_ep *ep,
 
 	if (err) {
 		FI_WARN(&sm2_prov, FI_LOG_EP_CTRL, "Error processing op\n");
+		xfer_entry->hdr.op_flags &= ~FI_DELIVERY_COMPLETE;
```

Comment:
I think it makes more sense to add this in the write_err_comp call in case it happens in the future. We should get rid of all the completion flags. You should probably add it in the complete_tx/rx paths so we don't report it over there either.